code/__HELPERS/botany.dm

```diff
@@ -0,0 +1,36 @@
+/// Tray Setters - The following macros adjust the hydroponics tray variables, and make sure that the stat doesn't go out of bounds.
+/**
+ * Adjust water.
+ * Raises or lowers tray water values by a set value. Adding water will dillute toxicity from the tray.
+ * * adjustamt - determines how much water the tray will be adjusted upwards or downwards.
+ */
+#define TRAY_ADJUST_WATER(tray, amt) tray.set_waterlevel(clamp(tray.waterlevel + amt, 0, tray.maxwater), FALSE)
```

Comment:
why are these all defines?

Reply:
Convenience. Since all they do is call a setter proc with a clamped value it'd suck to make them procs.

Reply:
Who for? Cause macros can be spooky for people, and it may harm understanding of your system. Unless it's so widely used or hot that it's a performance upgrade and all.